src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -830,8 +830,8 @@ private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
         }
 
         try {
-            StashManager.maybeClearAll(this);
-        } catch (IOException x) {
+            StashManager.maybeClearAll(this, /* or move up before closing getListener()? */new LogTaskListener(LOGGER, Level.FINE));
```

Comment:
Could have attempted to make such a change, but making a more conservative PR for now.